### PR TITLE
Unit tests must be done only at integration process

### PR DESCRIPTION
**Pull request by nerzhul**
test.cpp will be only compiled in Debug process and option is only available on debug build.

### CODE REVIEW

**Comment by nerzhul** [comment]
I fixed binary paths

**Comment by Zeno-** [comment]
I don't agree with this (at this point in time anyway). Without looking at the source code I am pretty sure unit tests are current run automatically when the build is debug but not when built as release; i.e. in release they are only run if the --enable-unittests flag is supplied at the command line.

As an example I have had one instance where I wanted to know if the user's unit tests passed but, like most of our users, they were using a pre-built binary of the release version but I could get the answer to my question because this switch is there.

I agree that it's probably not "normal" for unit tests to be available in a release build but it's not all that far fetched either. The unit tests are pretty minimal (at this time) and including them in the release build is no issue as far as I can tell. If unit tests become large (i.e. include more coverage) then //then// is the time to remove them from the release build and at the same time (and I'd argue that some kind of diagnostic mode be added at the same time -- probably a subset of the unit tests for the reasons outlined above).

TL;DR; Why?

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -355,6 +355,14 @@ add_subdirectory(jthread)
 add_subdirectory(script)
 add_subdirectory(util)
 
+if(${CMAKE_BUILD_TYPE} STREQUAL "Debug")
```
Comment (Zeno-) [inline src/CMakeLists.txt]: Should probably include SemiDebug as well

**Comment by nerzhul** [comment]
Then @Zeno- you suggest to remove the NDEBUG test and let the do-unittests into all compilation modes ? I think we could let them but they should return when they are run because this is needed for integration progress.

Please tell me what you want to to modify before merge and for merge

**Comment by Zeno-** [comment]
I want unit tests left in. I don't want NDEBUG defined for all build types of course.

But, NDEBUG is pretty much a useless macro at this point in time because assert(), in minetest at this time, assert() is misused. It's misused so many times that there is a special redefinition of it so that it doesn't go away in debug builds... that's the bigger problem, not unit tests being available as an option

**Comment by nerzhul** [comment]
unit tests are now available in all modes. But they only run when called by do-unittests and return a proper exit code for testing them in integration

**Comment by nerzhul** [comment]
b3aeba684b4e27e9fdf2c07949750bb567e40d6b ok @Zeno-
